src/tran.c

```diff
@@ -0,0 +1,353 @@
+        return point_adjust(point, E, N);
+    }
+
+    /* Then try nautical D:M.mm with GeographicLib-style colon separators */
+    if (7==sscanf (buf->buf, "%d:%lf%c %d:%lf%c %lf", &d, &ms, &E, &dd, &mmss, &N, &z)) {
+        point.lpz.lam = (d  + ms / 60);
+        point.lpz.phi = (dd + mmss/60);
+        point.lpz.z   = z;
+        buf->format = 4;
+        if (q)
+            *q = 'E';
+        return point_adjust(point, E, N);
+    }
+
+    /* Then try decimal degrees without northing/easting indicators */
+    if (3==sscanf (buf->buf, "%lf %lf %lf", &ms, &mmss, &z)) {
+        point.lpz.lam = ms;
+        point.lpz.phi = mmss;
+        point.lpz.z   = z;
+        buf->format = 5;
+        return point_adjust(point, 0, 0);
+    }
+
+    /* Finally try decimal degrees with northing/easting indicators */
+    if (5==sscanf (buf->buf, "%lf%c%lf%c%lf", &ms, &E, &mmss, &N, &z)) {
+        point.lpz.lam = ms;
+        point.lpz.phi = mmss;
+        point.lpz.z   = z;
+        buf->format = 6;
+        if (q)
+            *q = 'E';
+        return point_adjust(point, E, N);
+    }
+    return point;
+}
+
+
+char *skip_columns (char *buf, int howmany) {
+    int i;
+    for (i = 0;  i < howmany;  i++) {
+        while (isspace(*buf))
+            buf++;
+        while ((0!=*buf) && !isspace(*buf))
+            buf++;
+    }
+    return buf;
+}
+
+
+int main(int argc, char **argv) {
+    PJ *P;
+    int i, j, k, first_file_arg, direction;
+    FILE *in = 0;
+    struct buf buf = {0, 0, 0, 0, 0};
+    int angular_input = 1, angular_output = 0;
+
+    /* Used in handling classic '-'-style options */
+    char *opt_arg[256];
+    char takes_arg[256];
+
+    /* Used in handling '+'-style proj options */
+    #define MAX_PLUS_ARG 200
+    char *plus_argv[MAX_PLUS_ARG];
+    int   plus_argc = 0;
+
+    for (i = 0; i < 256; i++) {
+        opt_arg[i]   = 0;
+        takes_arg[i] = 0;
+    }
+
+    takes_arg[(int) 't'] = 1;
+    takes_arg[(int) 'o'] = 1;
+
+    /* Process all '-'-style options */
+    for (i = 1;  i < argc;  i++) {
+        int arg_group_size = strlen(argv[i]);
+
+        if ('-' != argv[i][0])
+            break;
+
+        for (j = 1; j < arg_group_size; j++) {
+            char c =  argv[i][j];
+
+            /* Flag style option, i.e. taking no arguments */
+            if (0==takes_arg[(int) c]) {
+                opt_arg[(int) c] = argv[0];
+                continue;
+            }
+
+            /* Option arg separate */
+            if (0==argv[i][j+1]) {
+                if ((argc==i + 1) || '+'==argv[i+1][0])
+                    return 0*fprintf (stderr, "Bad or missing arg for option \"%c\"\n", c);
+                opt_arg[(int) c] = argv[i + 1];
+                i++;
+                break;
+            }
+
+            /* Option arg inline */
+            opt_arg[(int) c] = argv[i] + j + 1;
+            break;
+        }
+    }
+
+    /* Process all '+'-style options */
+    for (i = i; i < argc; i++) {
```

Comment:
No - actually correct albeit somewhat funky. It just goes on where '-' style option processing left. Have added comment clarifying this